python/cudf/dataframe/index.py

```diff
@@ -204,18 +340,21 @@ def __getitem__(self, index):
             raise ValueError(index)
 
     def __eq__(self, other):
+        return super(RangeIndex, self).__eq__(other)
+
+    def equals(self, other):
         if isinstance(other, RangeIndex):
             return (self._start == other._start and self._stop == other._stop)
         else:
-            return super(RangeIndex, self).__eq__(other)
+            return (self == other)._values.all()
```

Comment:
I don't understand why you call `Index.__eq__` in one instance and create this `equals` method with other complicated logic. Shouldn't they do the same thing?

Reply:
`Index.__eq__` does an elementwise equals, so the return will be an Index of booleans. `equals` uses the elementwise equals and then gets the minimum value to determine a scalar Boolean type to return.

Reply:
Oh good distinction. 👍